src/lib/plot/series/mark-series.js

```diff
@@ -64,12 +71,14 @@ class MarkSeries extends AbstractSeries {
 
   _updateSeries() {
     const container = getDOMNode(this.refs.container);
-    const {data} = this.props;
+    const {
+      data,
+      dataKey} = this.props;
     if (!data) {
       return;
     }
     const circles = d3.select(container).selectAll('circle')
-      .data(data)
+      .data(data, dataKey)
```

Comment:
Actually, is it better to just assume that if there's an ID property it should be used as the key?

```js
const keyFunc = (data[0] && data[0].id) ? ((d, i) => (d && d.id) || i) : undefined;
const circles = d3.select(container).selectAll('circle')
  .data(data, keyFunc)
```